### PR TITLE
Remove documentation workflows

### PR DESCRIPTION
**Pull request by matthewwalsh0**
Remove the unnecessary documentation workflows.

Update the codeowners to the confirmations system team.

### CODE REVIEW

File: .github/CODEOWNERS
Comment (dbrans) [inline .github/CODEOWNERS:4]: I noticed that `MetaMask/devs` has been removed from the CODEOWNERS file. Could you provide some insight into the reason behind this change? I'm interested in understanding how this will impact the approval process for changes.

Comment (matthewwalsh0) [reply .github/CODEOWNERS:4]: We're trying to move away from broad ownership of repositories and instead assign them to specific teams.

If the repository is configured correctly in GitHub, this codeowner file should mean that all PRs need an approval from someone in our team before merging.

**Review by dbrans — APPROVED** [review]

**Pull request merged**
